api/handlers.go

```diff
@@ -495,16 +495,42 @@ func (si *ServerImplementation) SearchForTransactions(ctx echo.Context, params g
 		return badRequest(ctx, err.Error())
 	}
 
+	// Used for filtering duplicates after getting results.
+	rootTxnDedupeMap := make(map[string]struct{})
+	if filter.NextToken != "" {
+		_, _, txid, err := idb.DecodeTxnRowNext(filter.NextToken)
+		if err != nil {
+			badRequest(ctx, "bad next token")
+		}
+		rootTxnDedupeMap[txid] = struct{}{}
+	}
```

Comment:
Bootstrapping the dedupe map with NextToken solves the duplicate-root-txn-while-paging problem.